### PR TITLE
TMA-898: users brick deletes users from domain

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

File: lib/gooddata/lcm/actions/synchronize_users.rb [hunk 1]
```diff
@@ -183,6 +184,11 @@ def call(params)
           results = case mode
                     when 'add_to_organization'
                       domain.create_users(new_users.uniq { |u| u[:login] || u[:email] })
+                    when 'remove_from_organization'
+                      user_ids = new_users.uniq { |u| u[:login] || u[:email] }.map { |u| u[:login] || u[:email] }
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Calls 'new_users.uniq { |u| u[:login] || u[:email] }' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Calls 'new_users.uniq' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Calls 'u[:email]' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Calls 'u[:login]' 4 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Has the variable name 'u' (UncommunicativeVariableName)

File: lib/gooddata/lcm/actions/synchronize_users.rb [hunk 2]
```diff
@@ -183,6 +184,11 @@ def call(params)
...
+                      users = user_ids.map { |u| domain.users(u, client: client) }
+                      params.gdc_logger.warn "Deleting #{users.count} users from domain #{domain_name}"
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Calls 'params.gdc_logger' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Calls 'users.count' 3 times (DuplicateMethodCall)

File: lib/gooddata/lcm/actions/synchronize_users.rb [hunk 3]
```diff
@@ -183,6 +184,11 @@ def call(params)
...
+                      users.map { |u| u.delete }
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Style/SymbolProc: Pass `&:delete` as an argument to `map` instead of a block.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/45/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/858/ : FAILURE in 2m 30s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1126/ : SUCCESS in 1m 58s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/46/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/859/ : SUCCESS in 33s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1127/ : SUCCESS in 2m 14s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/47/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/860/ : SUCCESS in 33s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1128/ : SUCCESS in 32s

**Comment by Hahihula** [comment]
ok to merge

**Pull request merged**